rotkehlchen/chain/bitcoin/utils.py

```diff
@@ -107,21 +188,91 @@ def pubkey_to_p2sh_p2wpkh_address(data: bytes) -> BTCAddress:
 
 def pubkey_to_bech32_address(data: bytes, witver: int) -> BTCAddress:
     """
-    Bitcoin pubkey to bech32 address
+    Bitcoin pubkey to native Segwit(P2WPKH) & Taproot(P2TR) addresses.
 
     Source:
     https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#witness-program
-    https://github.com/mcdallas/cryptotools/blob/master/btctools/address.py
+    https://github.com/bitcoin/bips/blob/master/bip-0086.mediawiki
 
     May raise:
     - EncodingError if address could not be derived from public key
     """
-    witprog = hash160(data)
-    result = bech32.encode('bc', witver, witprog)
-    if not result:
-        raise EncodingError('Could not derive bech32 address from given public key')
+    witprog = hash160(data) if witver == 0 else _tweak_public_key(data)
+    spec = BitcoinSegwitEncoding.BECH32 if witver == 0 else BitcoinSegwitEncoding.BECH32M
+    converted_bits = bech32.convertbits(witprog, 8, 5)
+    if converted_bits is None:
+        raise EncodingError('Could not derive Bech32 address from given public key')
+    ret = _bech32_encode([witver] + converted_bits, spec)
+    if _bech32_decode(ret) == (None, None):
+        raise EncodingError('Could not derive Bech32 address from given public key')
+    return BTCAddress(ret)
+
+
+def _tweak_public_key(data: bytes) -> bytes:
```

Comment:
I don't think that's a common pattern across the codebase. But yeah, you're right

Reply:
Hmm we use libraries where we can. Just copy pasting code is dangerous since it can be full of errors, and we need to maintain it ourselves.

Can't you use an external library for most of this?

Reply:
> Hmm I think you need to find a library for this. Makes no sense to copy paste code like this for important bitcoin functionality like this. We should not have to maintain this code ourselves. It can't be that there is no python library for bitcoin that offers this?

Used [bip-utils](https://github.com/ebellocchia/bip_utils). 
Most of its dependencies are already used in the codebase. The only addition are: 
```
cbor2~=5.1
crcmod~=1.7
ecdsa~=0.15
```